solvation_analysis/analysis.py

```diff
@@ -0,0 +1,342 @@
+
+    @staticmethod
+    def _plot_solvation_radius(bins, data, radius):
+        """
+        Will plot the solvation radius on the rdf from bins, data, and a radius.
+        Includes a vertical line at the radius of interest.
+
+        Parameters
+        ----------
+            bins : np.array
+                the rdf bins
+            data : np.array
+                the rdf data
+            radius : float
+                the cutoff radius to draw on the plot
+
+        Returns
+        -------
+            Matplotlib Figure, Matplotlib Axes
+        """
+        fig, ax = plt.subplots()
+        ax.plot(bins, data, "b-", label="rdf")
+        ax.axvline(radius, color="r", label="solvation radius")
+        ax.set_xlabel("Radial Distance (A)")
+        ax.set_ylabel("Probability Density")
+        ax.legend()
+        return fig, ax
+
+    def plot_solvation_radius(self, res_name):
+        """
+        Will plot the rdf of a solvent molecule, specified by resname.
+        Includes a vertical line at the radius of interest.
+
+        Parameters
+        ----------
+        res_name : str
+            the name of the residue of interest, as written in the solvents dict
+
+        Returns
+        -------
+            Matplotlib Figure, Matplotlib Axes
+        """
+        bins, data = self.rdf_data[res_name]
+        fig, ax = self._plot_solvation_radius(bins, data, self.radii[res_name])
+        ax.set_title(f"Solvation distance of {res_name}")
+        return fig, ax
+
+    def radial_shell(self, solute_index, radius, step=None):
+        """
+        Returns all molecules with atoms within the radius of the central species.
+        (specifically, within the radius of the COM of central species).
+        Thin wrapper around solvation.get_radial_shell
+
+        Parameters
+        ----------
+            solute_index : int
+                the index of the solute of interest
+            radius : float or int
+                radius used for atom selection
+            step : int, optional
+                the step in the trajectory to perform selection at. Defaults to the
+                current trajectory step.
+
+        Returns
+        -------
+            AtomGroup
+        """
+        if step is not None:
+            self.u.trajectory[step]
+        return get_radial_shell(self.solute[solute_index], radius)
+
+    def closest_n_mol(self, solute_index, n_mol, step=None, **kwargs):
+        """
+        Returns the closest n molecules to the central species. Optionally returns
+        an array of their resids and an array of the distance of the closest atom
+        in each molecule. Thin wrapper around solvation.get_closest_n_mol, see
+        documentation for more detail.
+
+        Parameters
+        ----------
+            solute_index : Atom, AtomGroup, Residue, or ResidueGroup
+            n_mol : int
+                The number of molecules to return
+            step : int, optional
```

Comment:
frame is better. 

Ahh, if that's the spirit of MDA then this is definitely unsafe. Solutions would be:
1. remove the frame argument
pros: simple, consistent with rest of MDA
2. save the initial frame, change to new frame, perform selection, return to initial frame
pros: consistent with `solvation_shell` interface. compare selection methods without the hassle of changing frames.

My thinking is that users will be using solution.speciation to identify specific solute atoms and time steps of interest. It might be convenient to be able to query by solute_index and step without manually changing the trajectory step.

Reply:
I've implemented option 2 for now because I wanted to get something out. Still torn on this though so any input would be appreciated.

Reply:
Personally I'd go with the former (just because it avoid the extra code / need to check that you've gone back to the right frame / avoids changing coordinates if users are temporarily modifying them).

@hmacdope @richardjgowers might have different views here.

> My thinking is that users will be using solution.speciation to identify specific solute atoms and time steps of interest. It might be convenient to be able to query by solute_index and step without manually changing the trajectory step.

Do you have a general idea of how that code would look like, it might help with the discussion here?

Reply:
I'm imagining using it for visualization in particular, where I often want to select just a handful of molecules. For interactive visualization, I find it an inconvenience to manually manage the step in the trajectory. Additionally, by allowing a frame argument it keeps a more similar interface between `solvation_shell`, `radial_shell`, and `closest_n_mol`.

"""python
solution = Solution(solute, solvents)
shells_of_interest = solution.speciation.find_shells({"mol1": 3, "mol2": 3})
print(shells_of_interest)
"""python
*oh look, solute 1 in frame 4 has a solvation shell I am interested in, wonder what that looks like*
"""python
cool_shell = solution.solvation_shell(1, 4)
visualize_somehow(cool_shell)
"""
hmm, I'd like to see what the surrounding area looks like
"""python
bigger_shell = solution.radial_shell(1, 5, frame=4)
visualize_somehow(bigger_shell)
"""

That's how I'd imagine using it. Hope that's clear-ish. To be fair, I am only avoiding one line of code, `solution.u.trajectory[4]`, so not sure what's best. Happy to remove the frame keyword.

Reply:
@orioncohen I concur with @IAlibay, here (option 1, removing frame keyword). Even though it is a bit more clunky, consistency of interface and avoiding moving between frames internally is a big plus.

I could also see switching between frames internally going wrong in unexpected ways when paired with other analyses etc etc.

Also all of the functionality you have proposed here looks great, but is still perfectly fine without the frame keyword, indeed, it should be less code AFAIK.

Reply:
Alright! I'll move forward with no frame keyword. Thanks for the input y'all.

Reply:
>  To be fair, I am only avoiding one line of code, solution.u.trajectory[4], so not sure what's best.
I think you've answered your question here 🙃 

If the question is between implementing a complex / error-prone solution or letting users do a tiny bit more work, I'd always go for the latter. You can add a tutorial that show folks how to use it this way instead.